src/sentry/deletions/base.py

```diff
@@ -225,7 +225,7 @@ def chunk(self) -> bool:
                 return False
 
             self.delete_bulk(queryset)
-            remaining = remaining - query_limit
+            remaining = remaining - len(queryset)
```

Comment:
We have been scheduling an extra task even when there's nothing else to be deleted (see #91005).

This is why the extra scheduled task would fail to fetch a group since it was already deleted (`first_group = Group.objects.get(id=object_ids[0])`)